src/Support/RootField.php

```diff
@@ -194,6 +197,24 @@ public function abstractResolver($root, array $args, $context, ResolveInfo $info
 
         $args = new Arguments($args);
 
+        if (method_exists($this, 'rules')) {
+            $rules = $this->rules();
+            $messages = method_exists($this, 'messages') ? $this->messages() : [];
+            $attributes = method_exists($this, 'attributes') ? $this->attributes() : [];
+            $validator = Validator::make($args->toArray(), $rules, $messages, $attributes);
+
+            if ($validator->fails()) {
+                throw new ValidationException($validator->getMessageBag());
+            }
+        }
+
+        if (method_exists($this, 'authorize')) {
```

Comment:
Can we move these to own methods? Something like

```php
public function abstractResolver(...)
{
    //...

    if ( ! $this->passesAuthorization()) {
        throw new AuthorizationException();
    }

    //...
}

protected function passesAuthorization(): bool
{
    if (method_exists($this, 'authorize')) {
        return $this->container->call([$this, 'authorize']);
    }

    return true;
}
```

Same for a `passesValidation`.